http-client/src/main/java/io/micronaut/http/client/FilterResolver.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.micronaut.http.client;
+
+import io.micronaut.core.annotation.AnnotationMetadata;
+import io.micronaut.core.annotation.AnnotationMetadataResolver;
+import io.micronaut.core.util.ArrayUtils;
+import io.micronaut.core.util.PathMatcher;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.core.util.Toggleable;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.FilterProperties;
+import io.micronaut.http.filter.HttpClientFilter;
+import io.micronaut.http.filter.RuntimeHttpClientFilter;
+
+import java.lang.annotation.Annotation;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * Resolver that responsible to finding all the http client filters that should be applied to given request.
+ */
+class FilterResolver {
```

Comment:
If this is package private please mark as `@Internal`. However it seems to me that this is the type that should instantiated and passed in from the introduction advice

Reply:
Yeap, will change so that `FilterResolver` got injected from Introduction advice